### DISCUSSION TITLE
3-tier deployment in Azure?

### DISCUSSION THREAD

**Discussion by AlanSimmons** [uwrit/leaf#393] [Q&A]
Rush University Medical Center plans to deploy Leaf in its Azure environment. Our cybersecurity policies recommend one server per VM, so we would like to deploy the app server and web server on separate VM hosts--i.e., a 3-tier environment on Windows VMs hosted in Azure.

The installation documentation describes either a 3-tier configuration with Apache and Linux or a 2-tier configuration for Windows, in which the app and web servers are combined on one machine. We're facing challenges in trying to get the web and app servers to talk to each other when they are on separate Azure VMs. As I understand, the Leaf/Apache instructions can't help here, because we have to use .net.

One of our colleague institutions reports that they deployed in a 3-tier configuration using Windows machines, but using something other than Azure as the environment. (I think Netware.)

Is there a reason for the recommendation to combine the app and web servers in a Windows configuration? If not, are there recommendations for configuring the web and app servers so that they communicate? 

If we can get this to work with Windows VMS in Azure, we'll certainly send along how we did it. If we can't get this to work, I'll have to ask our cybersecurity team for an exception to the "one VM per tier" rule. 

Alan Simmons
Director, Clinical Research Analytics
Rush University Medical Center

**Comment by ndobb** [comment]
Thanks for your question, @AlanSimmons!  That makes sense.

The reason we recommend using a 2-tier IIS configuration in the Leaf docs is simply because our experience at UW has been using that model, as opposed that that being the only technical possibility. My recollection is that we briefly experimented with 3-tier deployments on IIS but had difficulty implementing a reverse-proxy in IIS to a secondary server hosting the API. 

I'm certain that a 3-tier deployment is possible (and if you or others are able to get that to work we're happy to add to the docs), but as we've never done so, we'd not be able to guide or help others troubleshoot.

Thanks again for letting us know and please let me know if you have any other questions.

Best,
-nic
